tap_xero/__init__.py

```diff
@@ -68,7 +68,7 @@ def ensure_credentials_are_valid(config):
     XeroClient(config).filter("currencies")
 
 def discover(ctx):
-    ctx.refresh_credentials()
+    ctx.check_platform_access()
```

Comment:
Code added w.r.to TDL-594 (Check authentication and authorization during discovery mode)